### PR TITLE
[release/8.1] Enable full signing and stable package versions

### PR DESCRIPTION
**Pull request by joperezr**
Getting ready for 8.1 release.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4989)

### CODE REVIEW

File: src/Aspire.Hosting.Azure.WebPubSub/Aspire.Hosting.Azure.WebPubSub.csproj
Comment (eerhardt) [inline src/Aspire.Hosting.Azure.WebPubSub/Aspire.Hosting.Azure.WebPubSub.csproj:10]: FYI @vicancy

**Review by eerhardt — APPROVED** [review]

File: src/Directory.Build.targets
Comment (radical) [inline src/Directory.Build.targets:8]: Just curious, why is this change needed?

**Comment by radical** [comment]
Is this branch meant for `release/8.0` instead of `main`?

File: src/Directory.Build.targets
Comment (joperezr) [reply src/Directory.Build.targets:8]: Because not all packages are going stable, so given this PR is stabilizing package versions, some will be like: `8.1.0` and some other swill be like `8.1.0-ci`. This change is ensuring that each project is returning the actual package version it will produce, as opposed to just assuming that all will have the same one.

**Comment by joperezr** [comment]
Ohh great catch, yes retargeting.

File: src/Directory.Build.targets
Comment (radical) [reply src/Directory.Build.targets:8]: Ah, good idea!!

File: tests/Aspire.EndToEnd.Tests/Aspire.EndToEnd.Tests.csproj
Comment (eerhardt) [inline tests/Aspire.EndToEnd.Tests/Aspire.EndToEnd.Tests.csproj:5]: Why do we need to hard code this? Why can't it be generated from the other MSBuild properties?

Comment (joperezr) [reply tests/Aspire.EndToEnd.Tests/Aspire.EndToEnd.Tests.csproj:5]: I tried locally using major and minor version but I think this project didn't import the right props and due to timing just wanted to unblock the build. End to End tests are failing now, so I guess now I have to fix than anyway.

Comment (joperezr) [reply tests/Aspire.EndToEnd.Tests/Aspire.EndToEnd.Tests.csproj:5]: Looks like there are several assumptions that the testing infrastructure makes around package versions which may not be correct as they assume that the same package version will be used everywhere, which will certainly not always be the case. I just spent some time trying to fix another package testing issue I was seeing, but couldn't get the versions to flow correctly so due to time I just pushed another change which hardcodes the versions for now to unblock the build and be able to get a build out for validation. We should of course undo these hard coded hacks, but we should also fix these assumptions on versions. cc: @radical

Comment (eerhardt) [reply tests/Aspire.EndToEnd.Tests/Aspire.EndToEnd.Tests.csproj:5]: Do we have an issue tracking this to fix it in `main`?

**Comment by eerhardt** [comment]
@joperezr - Are you going to forward port any relevant changes here to `main`?

**Comment by joperezr** [comment]
After release, we always merge back the release branch into main in order to make sure that changes that were done are not missed.

**Pull request merged**
